kafka-client/build.gradle.kts

```diff
@@ -16,8 +18,79 @@
 
 plugins {
     id("responsive.java-library-conventions")
+    id("java")
+}
+
+/*********** Generated Resources ***********/
+
+val gitCommitId: String by lazy {
+    val stdout = ByteArrayOutputStream()
+    //rootProject.exec {
+    exec {
+        commandLine("git", "rev-parse", "--verify", "--short", "HEAD")
+        standardOutput = stdout
+    }
+    stdout.toString().trim()
+}
+
+val writeVersionPropertiesFile = "writeVersionPropertiesFile"
+val gitVersion = version
+
+val resourcesDir = "$buildDir/resources/main"
+val versionFilePath = "$resourcesDir/version.properties"
+
+tasks.register(writeVersionPropertiesFile) {
+    val versionFile = file(versionFilePath)
+    outputs.file(versionFile)
+    doLast {
+        file(versionFilePath).writeText(
+                "git.build.version=" + gitVersion + "\n" +
+                "git.commit.id=" + gitCommitId + "\n"
+        )
+    }
 }
 
+// Need to add all generated resource directories as outputs of the main sourceSet
+sourceSets {
+    main {
+        output.dir(file("$buildDir/resources"))
+    }
+}
+
+
+// TODO: there has GOT to be a more concise way to do this than listing it out for every task individually...
```

Comment:
I think these all depend on either compileJava or compileTestJava - we should be able to put it just in those two (I don't know whether compileTestJava depends on compileJava or not, I think it probably does)

Reply:
Yeah I had that originally and could not for the life of me figure out why it was still complaining -- turns out I had added something to the build file that needed to be explicitly defined for each task, hence this whole mess, but wasn't necessary to begin with. Anyways reduced it down to just the compileJava task